### PR TITLE
Feat: Add filters to Dom Events tab

### LINKED ISSUE #38010: DOM Events tab should be searchable

### PR DESCRIPTION
**Pull request by jesus4497**
Added filters to the Dom Events tabs. Now the user can filter by event type or search for a word that could be in the html value. Closes #38010 

https://user-images.githubusercontent.com/39612839/187308139-2d402ffa-cd06-4614-bf6e-21c68d4d15c0.mp4

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: static/app/views/replays/detail/domMutations/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        const filter = filters[key];
+        if (!filter(mutation)) {
+          return false;
+        }
+      }
+    }
+
+    return doesMatch;
+  });
+};
```
Comment (ryan953) [inline static/app/views/replays/detail/domMutations/utils.tsx:38]: can any parts of this file be shared with the network filter/utils?

Comment (jesus4497) [reply static/app/views/replays/detail/domMutations/utils.tsx]: Made the whole file generic, so it can be shared with the network filters!

**Review by ryan953 — APPROVED** [review]

File: static/app/views/replays/detail/domMutations/index.tsx [hunk 2]
```diff
@@ -1,67 +1,153 @@
...
+        <SearchBar size="sm" onChange={handleSearch} placeholder={t('Search DOM')} />
+      </MutationFilters>
+      <MutationList>
+        {filteredDomMutations.map((mutation, i) => (
+          <MutationListItem
+            key={i}
+            onMouseEnter={() => handleMouseEnter(mutation.crumb)}
+            onMouseLeave={() => handleMouseLeave(mutation.crumb)}
+          >
+            {i < actions.length - 1 && <StepConnector />}
```
Comment (ryan953) [inline static/app/views/replays/detail/domMutations/index.tsx:107]:
```suggestion
            {i < filteredDomMutations.length - 1 && <StepConnector />}
```

**Comment by Jesse-Box** [comment]
@jesus4497 The comment I made in #38165 apply here also.

**Pull request merged**